File: backend/src/models/candidate/candidate.model.ts
import mongoose, { Model, Schema } from "mongoose";
import type { ICandidate } from "./candidate.interface.js";
import validator from "validator";

const candidateSchema = new Schema<ICandidate>(
  {
    name: {
      type: String,
      required: true,
      trim: true,
    },
    email: {
      type: String,
      required: true,
      unique: true,
      lowercase: true,
      trim: true,
      validate: [validator.isEmail, "Please provide a valid Email address"],
    },
    phone: { type: String, default: "" },
    linkedInUrl: { type: String, default: "" },
    githubUrl: { type: String, default: "" },
    location: { type: String, default: "" },
    resumeUrl: { type: String, required: true },
  },
  { timestamps: true },
);

export const Candidate = mongoose.model<ICandidate>(
  "Candidate",
  candidateSchema,
);

File: ai-services/app/models/candidate_model.py
from typing import Optional
from beanie import Document
from pydantic import Field

class Candidate(Document):
    name: str
    email: str
    phone: Optional[str] = None
    location: Optional[str] = None
    githubUrl: Optional[str] = None
    linkedInUrl: Optional[str] = None
    latestResumeUrl: Optional[str] = None

    class Settings:
        name = "candidates"